internal/worker/sshsession/worker_test.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshsession_test
+
+import (
+	"bytes"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/worker/v3/workertest"
+	"go.uber.org/mock/gomock"
+	"golang.org/x/crypto/ssh"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/internal/worker/sshsession"
+	"github.com/juju/juju/rpc/params"
+)
+
+type workerSuite struct {
+	testing.IsolationSuite
+
+	facadeClientMock        *MockFacadeClient
+	watcherMock             *MockStringsWatcher
+	ephemeralkeyUpdaterMock *MockEphemeralKeysUpdater
+}
+
+var _ = gc.Suite(&workerSuite{})
+
+func (s *workerSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+
+	s.facadeClientMock = NewMockFacadeClient(ctrl)
+	s.watcherMock = NewMockStringsWatcher(ctrl)
+	s.ephemeralkeyUpdaterMock = NewMockEphemeralKeysUpdater(ctrl)
```

Comment:
For naming, I'd remove the word "Mock" to stick with convention across other files/tests.